src/components/Event/Event.st.css

```diff
@@ -0,0 +1,143 @@
+:import {
+    -st-from: "../../common/formatters.st";
+    -st-named: color, applyOpacity, fallback, font;
+}
+
+:import {
+    -st-from: "../Text/Text.st.css";
+    -st-default: TPAText;
+}
+
+/*Defaults*/
+:vars {
+    DefaultFontStyle: "{theme: 'Body-M', size: '14px', lineHeight: '1em'}";
+    DefaultTimeColor: color-4;
+    DefaultTitleColor: color-5;
+    DefaultAllDayBackground: color-8;
+    DefaultAllDayOpacity: 0.2; 
+    DefaultTopPadding: 4px;
+    DefaultRightPadding: 4px;
+    DefaultBottomPadding: 4px;
+    DefaultLeftPadding: 12px;
+    DefaultRTLTopPadding: 4px;
+    DefaultRTLRightPadding: 12px;
+    DefaultRTLBottomPadding: 4px;
+    DefaultRTLLeftPadding: 4px;
+    BeforeColor: color(color-4);
+    BeforeOpacity: 0.06;
+}
+
+/*Overrides*/
+:vars {
+    FontStyle: --overridable;
+    TimeColor: --overridable;
+    TitleColor: --overridable;
+    AllDayBackground:  --overridable;
+    AllDayOpacity: --overridable;
+    TopPadding: --overridable;
+    RightPadding: --overridable;
+    BottomPadding: --overridable;
+    LeftPadding: --overridable;
+    RTLTopPadding: --overridable;
+    RTLRightPadding: --overridable;
+    RTLBottomPadding: --overridable;
+    RTLLeftPadding: --overridable;
+}
+
+/* Fallbacks */
+:vars {
+    CurrentFontStyle: fallback(value(FontStyle), font(value(DefaultFontStyle)));
+    CurrentTimeColor: color(fallback(value(TimeColor), value(DefaultTimeColor)));
+    CurrentTitleColor: color(fallback(value(TitleColor), value(DefaultTitleColor)));
+    CurrentAllDayBackground: color(fallback(value(AllDayBackground), value(DefaultAllDayBackground)));
+    CurrentAllDayOpacity: fallback(value(AllDayOpacity), value(DefaultAllDayOpacity));
+    CurrentTopPadding: fallback(value(TopPadding),value(DefaultTopPadding));
+    CurrentRightPadding: fallback(value(RightPadding),value(DefaultRightPadding));
+    CurrentLeftPadding: fallback(value(LeftPadding),value(DefaultLeftPadding));
+    CurrentBottomPadding: fallback(value(BottomPadding),value(DefaultBottomPadding));
+    CurrentRTLTopPadding: fallback(value(RTLTopPadding),value(DefaultRTLTopPadding));
+    CurrentRTLRightPadding: fallback(value(RTLRightPadding),value(DefaultRTLRightPadding));
+    CurrentRTLLeftPadding: fallback(value(RTLLeftPadding),value(DefaultRTLLeftPadding));
+    CurrentRTLBottomPadding: fallback(value(RTLBottomPadding),value(DefaultRTLBottomPadding));
+}
+
+.root {
+    -st-states: isMultiday, isSelected, isRightToLeft;
+    position: relative;
+    padding: value(CurrentTopPadding) value(CurrentRightPadding) value(CurrentBottomPadding) value(CurrentLeftPadding);
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+}
+
+.root:isRightToLeft {
```

Comment:
If we really need to pass an `rtl` state from javascript, then we use the `TPAComponentsConsumer`.
You can see many examples for using this in the code.
But before we use this, we prefer having a `direction` agnostic approach.
Meaning, using flex, and other solutions (`::before`, `::after`).

Reply:
Flex was the first thing i try, and it is the better practice. It's just that the shadowing requirement force the component to be relative positioned because the `:before` element needs to be absolute in order to be on top of the event. so  flex didn't work. I actually wanted to talk to you about it. You have an idea what could be done to achieve shadowing with no abstract ? I think that if we could figure this one out flex would work

Reply:
what do you mean by `shadowing`?

Reply:
on hover and when selected there should be shadow div. like black but transparent. you can view it here: https://app.abstract.com/projects/3c6113bf-5996-4c28-aaa3-14ab8b599eb4/branches/master/commits/latest/files/7997d7f2-7bd7-4fe3-9e22-017d7c436e04/layers/330883A3-2E9F-45E5-9AFE-DEE02851C9B2?collectionId=117a41e8-de09-499c-b98a-daa1409657c2&collectionLayerId=8ceb6c0c-c21a-42b4-bc70-48e7aeedbb37